feat(docs): add new local CLI docs (#18502)
expo/expo: docs/pages/workflow/legacy-expo-cli.md

---
title: Legacy Global Expo CLI

> maxHeadingDepth: 4

Author: Is there a better place for deprecated docs?

Reviewer: Hmm, maybe https://docs.expo.dev/archive/ or "Deprecated" group at the end of guides (https://docs.expo.dev/guides/)?

Reviewer 2: You can't navigate to archive, and you still need the legacy CLI for some commands, so Deprecated section in guides seems better.

Reviewer 3: `pages/archive`

---
title: Legacy Global Expo CLI
maxHeadingDepth: 4
---

import { Terminal } from '~/ui/components/Snippet';

> The global Expo CLI `expo-cli` was succeeded by the local CLI `@expo/cli` in SDK 46. This doc contains info on the now deprecated `expo-cli` package.

Reviewer 4: ```suggestion
The global Expo CLI `expo-cli` was succeeded by the local CLI `@expo/cli` in SDK 46. This documentaion contains info on the now deprecated `expo-cli` package.
```

> If you're using a project that is SDK 46 or higher, please refer to the [Expo CLI docs](/workflow/expo-cli/).

Reviewer 4: ```suggestion
If you're using a project that is SDK 46 or higher, refer to the [Expo CLI docs](/workflow/expo-cli/).
```

## Installation

The legacy Expo CLI can be installed as a global npm package on your development machine. To download and install it, run the following command:

<Terminal cmd={['$ npm install -g expo-cli']} />

> Not versioning the CLI with your project lowers backwards compatibility, and makes it harder to use your project in continuos environments like GitHub Actions. To combat this, we moved to using a versioned CLI in SDK 46.

Reviewer 2: This was marked as resolved, but is not resolved.

## Checking CLI Version

To check the current version of global Expo CLI you are using, use the `--version` option.

<Terminal cmd={['$ expo --version']} />

## Commands

You can view the list of commands available with your version in your terminal using the `--help` option.

<Terminal cmd={['$ expo --help']} />

> To learn more about a specific command and its options, use `expo [command] --help`.

Reviewer 4: ```suggestion
To learn more about a specific command and the options it may have, use `expo [command] --help`.
```

  - Dropped arguments: `--public-url`, `--asset-url`, `--merge-src-url`, `--merge-src-dir`, `-t, --target`, `--experimental-bundle`, `--config`, `-q, --quiet`
  - Dropped aliases for `--dump-assetmap` (`-d`) and `--dump-sourcemap` (`-s`).
  - Dropped warning about clearing `dist` folder that is shown between runs, this is akin to all other bundler commands across web tooling.
  - Dropped all merging code because experimental bundle does not support index files so we don't need to support merging index files.
  - Dropped the quiet flag as this appears to do nothing anymore.
- Parts of `expo upgrade` and `expo doctor` have been merged into `npx expo install` via the `--fix` and `--check` flags. [PR](https://github.com/expo/expo/pull/17048).
- Dropped unused generated `.exprc`, `.expo/packager-info.json` files.
- Dropped all deprecated support for `*.expo.js` files.
- Reduced usage of `.expo/settings.json` to provide a more sandboxed state between processes.

#### Environment Variables

- Dropped `$XDL_PORT`, `$XDL_HOST`, `$XDL_SCHEME`, `$SERVER_URL` in favor of `$EXPO_STAGING` and `$EXPO_LOCAL` environment variables.

> - Dropped `$EXPO_PACKAGER_HOSTNAME`, `$EXPO_MANIFEST_PROXY_URL` environment variables.
> - Dropped `$EXPO_EDITOR` in favor of `$EDITOR` environment variable.
> - Utilize `$DEBUG=expo:*` more and `$EXPO_DEBUG` less (more control).

Author: ```suggestion
```

none of this is true anymore
